### PR TITLE
entry type:layer to replace mvt_clone

### PR DESCRIPTION
**Pull request by dbauszus-glx**
The mvt_clone type entry had several bugs and applied an unnecessary complex logic.

The mvt_clone module has been removed and the mvt_clone entry method has been deprecated. The mvt_clone method will warn and return the new layer entry method.

A `layer.featureSet` filter has been to the featureStyle method. Feature not included in the featureSet will not be styled.

The `layer/format/mvt` module documentation has been revised. The clone logic has been removed.

The `type:layer` entry method assigns a Key property which is the orignal layer key required to get a location.

The `highlight` interaction `getFeature` method has been updated to recognise the layer.Key property.

related fixes:

Removed `layer/featureStyle` module function params.

Added missing `@requires` to `/ui/locations/entries` module.

Fixed `/ui/locations/entries/numeric` declaration.

Fixed `/layer/changeEnd` declaration.

Removed deprecated ui/locations/entries/default module.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206602587669819

### CODE REVIEW

**Comment by dbauszus-glx** [comment]
The featureSet and lookupFeature logic has now been merged in the showLayer method of the layer entry and the featureProperties method of the featureStyle module.

The layer decoration in the layer entry has been moved to the decorateLayer method in the layer entry module.

A changeEnd event listener for zoom restricted layer has been added to the  decorateLayer to disable the entry checkbox if the entry layer is out of range.

https://github.com/GEOLYTIX/xyz/issues/1543 must be completed before the layer entry PR can be resolved.

File: lib/ui/locations/entries/layer.mjs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+A layer.view will be created for the decorated layer and appended to the entry.node for the location view.
+
+@param {infoj-entry} entry type:layer entry.
+@property {string} [entry.layer] lookup layer key for locale.layers[].
+
+@return {HTMLElement} Node element to hold the layer view drawer.
+*/
+export default function layer(entry) {
+
+  console.log(entry)
```
Comment (RobAndrewHurst) [inline lib/ui/locations/entries/layer.mjs:28]: This should be removed!
```suggestion
  console.log(entry)
```

Comment (RobAndrewHurst) [inline lib/ui/locations/entries/layer.mjs:28]: This should be removed! 
```suggestion
```

**Comment by RobAndrewHurst** [comment]
a4edfd93aaf4288ee3220343281cad8f58107fd0

```js
// Request the location fields from layer json & id.
  const response = await mapp.utils.xhr(`${mapp.host}/api/query?` + mapp.utils.paramString({
    template: location.getTemplate,
    locale: location.locale,
    layer: location.layer.key,
    table: location.table,
    id: location.id,
  }));
```

I noticed that the `layer` property on the `getInfoj()` method in the lib/location/get.mjs module will be set by a unique layer name from the layer entry, which wouldn't work. I am going to add in the `location.layer.Key` as a first option when querying the data.

```js
  // Request the location fields from layer json & id.
  const response = await mapp.utils.xhr(`${mapp.host}/api/query?` + mapp.utils.paramString({
    template: location.getTemplate,
    locale: location.locale,
    layer: location.layer.Key || location.layer.key,
    table: location.table,
    id: location.id,
  }));
```

**Review by simon-leech — CHANGES_REQUESTED** [review]
An `infoj` `entry` `layer` with multiple thematics is not working correctly. 

the theme is swapped correctly on the mapview, but the legend is not updated and the legend styling is just lost. 


To swap an `mvt_clone` to type `layer` - is it supposed to be as simple as just changing the type? 
I did this but I crash here 
image

**Comment by dbauszus-glx** [comment]
> An `infoj` `entry` `layer` with multiple thematics is not working correctly.
> 
> the theme is swapped correctly on the mapview, but the legend is not updated and the legend styling is just lost.
> 
> To swap an `mvt_clone` to type `layer` - is it supposed to be as simple as just changing the type? I did this but I crash here on layer.style.theme.title image

Yes, the layer entry type will replace the mvt_clone entry type method.

**Comment by dbauszus-glx** [comment]
https://github.com/GEOLYTIX/xyz/pull/1386/commits/b81cd41c394527d5355cb572246fb3b01aadac0f

@simon-leech The layer entry method will now assign the layer.style.panel and the dropdown callback will replace the children in the style.panel.

**Review by simon-leech — APPROVED** [review]
Working for me - tested using mvt_clone and layer method and both are working for me :)

**Review by RobAndrewHurst — APPROVED** [review]

**Pull request merged**